### PR TITLE
Minecraft bow

### PR DESCRIPTION
**Pull request by ghost**
Minecraft bow

### CODE REVIEW

File: ewcfg.py [hunk 1]
```diff
@@ -2016,7 +2017,7 @@
 weapon_id_pickaxe = 'pickaxe'
 weapon_id_bass = 'bass'
 weapon_id_umbrella = 'umbrella'
-
+weapon_id_bass = 'bow'
```
Comment (magicksRampage) [inline ewcfg.py]: should be weapon_id_bow

File: ewcfg.py [hunk 2]
```diff
@@ -3193,6 +3194,31 @@ def wef_umbrella(ctn = None):
...
+	ctn.slimes_damage = int(ctn.slimes_damage * 2.5)
+
+	if aim <= (-2 + int(13 * ctn.miss_mod)):
+		if mutation_id_sharptoother in user_mutations:
+			if random.random() < 0.5:
+				ctn.miss = True
+		else:
+			ctn.miss = True
+
+	elif aim >= (9 - int(16 * ctn.crit_mod)):
```
Comment (magicksRampage) [inline ewcfg.py:3218]: crit mod usually scales with the aim range, i.e. 13 in this case

File: ewcfg.py [hunk 3]
```diff
@@ -14164,7 +14228,7 @@ def get_strat_g(combat_data, in_range, first_turn, active):
...
 	weapon_id_garrote: "**The garrote wire** is a weapon for sale at the Dojo. Attacking with the garrote costs 5 sap. It has a damage mod of 15 and an attack cost mod of 1. It doesn't require a captcha and it pierces all enemy hardened sap. It has a 0% miss chance and a 1% chance for a crit, which does 10x damage. When you attack with a garrote, the target has 5 seconds to send any message before the damage is done. If they do, the attack fails.",
-	
+        weapon_id_bow: "The minecraft bow** is a weapon not for sale at the Dojo. Attacking with the bass costs 3 sap. It has a damage mod of 2.5 and an attack cost mod of 1. It has a miss chance of 1/13 and a 7/13 chance for a crit, which increases the damage mod to 4. The minecraft bow does not require a captcha to use. The minecraft has sap crushing 1 and sap piercing 8. If you takes less than 20 seconds between attacks, your miss chance will increase."
```
Comment (magicksRampage) [inline ewcfg.py]: typo: Attacking with the bow*

Comment (magicksRampage) [reply ewcfg.py]: in the code the soft cooldown is 10 seconds, not 20

Comment (magicksRampage) [reply ewcfg.py]: the crit chance is only 2/13 in the code, not 7/13

Comment (magicksRampage) [reply ewcfg.py]: sap cost is 2 in the code, not 3

**Review by magicksRampage — CHANGES_REQUESTED** [review]
the bow seems pretty underpowered. I'd recommend lowering the soft cooldown from 10 to 5 seconds

File: ewcfg.py [hunk 4]
```diff
@@ -14164,7 +14228,7 @@ def get_strat_g(combat_data, in_range, first_turn, active):
...
 	weapon_id_garrote: "**The garrote wire** is a weapon for sale at the Dojo. Attacking with the garrote costs 5 sap. It has a damage mod of 15 and an attack cost mod of 1. It doesn't require a captcha and it pierces all enemy hardened sap. It has a 0% miss chance and a 1% chance for a crit, which does 10x damage. When you attack with a garrote, the target has 5 seconds to send any message before the damage is done. If they do, the attack fails.",
-	
+        weapon_id_bow: "The minecraft bow** is a weapon not for sale at the Dojo. Attacking with the bow costs 2 sap. It has a damage mod of 4 and an attack cost mod of 1. It has a miss chance of 1/13 and a 2/13 chance for a crit, which increases the damage mod to 8. The minecraft bow does not require a captcha to use. The minecraft bow has sap crushing 1 and sap piercing 8. If you takes less than 10 seconds between attacks, your miss chance will increase."
```
Comment (magicksRampage) [inline ewcfg.py]: crit mod is 10 in the code now

File: ewcfg.py [hunk 5]
```diff
@@ -3193,6 +3194,31 @@ def wef_umbrella(ctn = None):
...
+	ctn.sap_damage = 1
+	ctn.sap_ignored = 8
+	
+	# Increased miss chance if attacking within less than two seconds after last attack
+	time_lastattack = ctn.time_now - (float(ctn.weapon_item.item_props.get("time_lastattack")) if ctn.weapon_item.item_props.get("time_lastattack") != None else ctn.time_now)
+	ctn.miss_mod += (((10 - min(time_lastattack, 10)) / 10) ** 2) / 13 * 10
+
+	ctn.slimes_damage = int(ctn.slimes_damage * 4)
+
+	if aim <= (-2 + int(16 * ctn.miss_mod)):
```
Comment (magicksRampage) [inline ewcfg.py]: why did you change this one to 16 instead of the other one to 13?

**Review by magicksRampage — CHANGES_REQUESTED** [review]

File: ewcfg.py (hunk 5 above)
Comment (ghost) [reply ewcfg.py]: Uh, is it bad if  i do that?
